check that currentScript is set by a script tag with optional chaining

lkuechler opened the pull request:
Based on this PR but using optional chaining: https://github.com/PrismJS/prism/pull/3863

LeaVerou reviewed (changes requested):
Hi @lkuechler, would you be able to repurpose this to be against v2? See comment on #3863

lkuechler committed:
check that currentScript is set by a script tag

Based on: https://gist.github.com/jackfromeast/aeb128e44f05f95828a1a824708df660
Should resolve: https://osv.dev/vulnerability/GHSA-x7hr-w5r2-h6wg

lkuechler commented:
@LeaVerou I also applied the same changed to the v2 version now. There where a few more cases but I think I got all of them
